### DISCUSSION TITLE
WebContainers

### DISCUSSION THREAD

**Discussion by nkovacic** [codesandbox/sandpack#54] [Ideas]
Running Node.js inside the browser is the holy grail for web-based IDE and there is [finally movement in this direction](https://github.com/stackblitz/webcontainer-core).

Are there any plans for codesandbox to incorporate this into sandpack or codesandbox-client?

**Comment by CompuIves** [comment]
Hey! WebContainers is super interesting, but I don't think it directly applies to Sandpack because Sandpack is a bundler instead of a Node.js environment. To CodeSandbox it would apply, and especially for the projects that need a Node environment on there (eg. Next.js) it would be super interesting, but we can only use it if WebContainers goes open source. At that point it's something that we can implement!

**Comment by dinbtechit** [comment]
Hey this is interesting. Just curious could we embed webcontainers in chrome browser extensions?
